src/client/java/dev/creesch/model/WebsocketMessageBuilder.java

```diff
@@ -196,4 +210,89 @@ public static WebsocketJsonMessage createHistoryMetaDataMessage(List<WebsocketJs
             minecraftVersion
         );
     }
+
+    private static final Pattern MINECRAFT_TEXTURE_URL_PATTERN = Pattern.compile("^https?://textures\\.minecraft\\.net/texture/.+");
+
+    private static String getPlayerTextureUrl(GameProfile profile) {
+        Collection<Property> textures = profile.getProperties().get("textures");
+        if (textures.isEmpty()) {
+            return "unknown";
+        }
+
+        // Generally there should only be one texture.
+        // But, it is apparently possible for modified servers and such to add multiple.
+        // Mojang signs textures, but that seems overkill.
+        // Instead, simply grab the first texture that is minecraft hosted. Fingers crossed there aren't multiple hostnames for textures.
+        for (Property property : textures) {
+            try {
+                String decodedValue = new String(Base64.getDecoder().decode(property.value()), StandardCharsets.UTF_8);
+                JsonObject textureJson = JsonParser.parseString(decodedValue).getAsJsonObject();
+                JsonObject texturesObj = textureJson.getAsJsonObject("textures");
+
+                if (texturesObj.has("SKIN")) {
+                    String textureURL = texturesObj.getAsJsonObject("SKIN")
+                    .get("url")
```

Comment:
Can I get a standard order of indentation on this line? (Also we should have an auto-formatter)

Reply:
Not sure if auto formatters for java are a thing, but I was meaning to at least set something like checkstyles up.